integrations/apify/src/actions/start-crawler-run.ts

```diff
@@ -0,0 +1,103 @@
+import * as bp from '.botpress';
+import { getClient } from 'src/client';
+ 
+export const startCrawlerRun = async ({ 
+  ctx, 
+  client,
+  input, 
+  logger
+}: {
+  ctx: bp.Context;
+  client: bp.Client;
+  input: { startUrls: string[]; excludeUrlGlobs?: string[]; includeUrlGlobs?: string[]; 
+    maxCrawlPages?: number; kbId: string; saveMarkdown?: boolean; 
+    htmlTransformer?: 'readableTextIfPossible' | 'readableText' | 'minimal' | 'none'; 
+    removeElementsCssSelector?: string; 
+    crawlerType?: 'playwright:adaptive' | 'playwright:firefox' | 'cheerio' | 'jsdom' | 'playwright:chrome'; 
+    expandClickableElements?: boolean; headers?: Record<string, string>; rawInputJsonOverride?: string; };
+  logger: bp.Logger;
+}) => {
+  logger.forBot().info(`Starting crawler run with input: ${JSON.stringify(input)}`);
+
+  try {
+    const apifyClient = getClient(
+      ctx.configuration.apiToken,
+      client,
+      logger
+    );
+
+    const params = {
+      startUrls: input.startUrls,
+      excludeUrlGlobs: input.excludeUrlGlobs,
+      includeUrlGlobs: input.includeUrlGlobs,
+      maxCrawlPages: input.maxCrawlPages,
+      saveMarkdown: input.saveMarkdown,
+      htmlTransformer: input.htmlTransformer,
+      removeElementsCssSelector: input.removeElementsCssSelector,
+      crawlerType: input.crawlerType,
+      expandClickableElements: input.expandClickableElements,
+      headers: input.headers,
+      rawInputJsonOverride: input.rawInputJsonOverride,
+      kbId: input.kbId,
+    } as const;
+
+    const result = await apifyClient.startCrawlerRun(params);
+
+    if (result.success) {
+      logger.forBot().info(`Crawler run started successfully. Run ID: ${result.data?.runId}`);
+      logger.forBot().debug(`Start result: ${JSON.stringify(result.data)}`);
+
+      const runId = result?.data?.runId;
+      if (runId) {
+        try {
+          logger.forBot().info(`Persisting kbId mapping for run ${runId} -> kb ${input.kbId}`);
+          const existing = await client.getState({
+            type: 'integration',
+            id: ctx.integrationId,
+            name: 'apifyRunMappings',
+          });
+
+          const payload = existing?.state?.payload;
+          const currentMap = payload || {};
+          currentMap[runId] = input.kbId;
+
+          await client.setState({
+            type: 'integration',
+            id: ctx.integrationId,
+            name: 'apifyRunMappings',
+            payload: currentMap,
+          });
+
+          logger.forBot().info(`Persisted kbId mapping for run ${runId}`);
+        } catch (stateError) {
+          logger.forBot().warn(`Failed to persist kbId mapping for run ${runId}: ${stateError instanceof Error ? stateError.message : String(stateError)}`);
```

Comment:
question: related to my other comment, since kbId is supposed to be a string, it looks like this only fails if there is a network error, right? In that case I think we should stop the execution of the script.